File: FlashCards-SPA/src/app/core/_models/_dtos/fromServer/flashcardForLearn.ts
export class FlashcardForLearn {
  flashcardId: number;
  flashcardSubscriptionId: number;
  phrase: string;
  phrasePronunciation: string;
  phraseSampleSentence: string;
  phraseComment: string;
  translatedPhrase: string;
  translatedPhraseSampleSentence: string;
  translatedPhraseComment: string;
  languageLevel: string;
  category: string;
  trainLevel: number;
  markedAsHard: boolean;
  lastTrainingDate: Date;

  public initialize(flashcardId: number,
                    flashcardSubscriptionId: number,
                    phrase: string,
                    phrasePronunciation: string,
                    phraseSampleSentence: string,
                    phraseComment: string,
                    translatedPhrase: string,
                    translatedPhraseSampleSentence: string,
                    translatedPhraseComment: string,
                    languageLevel: string,
                    category: string,
                    trainLevel: number,
                    markedAsHard: boolean,
                    lastTrainingDate: Date): void {
    this.flashcardId = flashcardId;
    this.flashcardSubscriptionId = flashcardSubscriptionId;
    this.phrase = phrase;
    this.phrasePronunciation = phrasePronunciation;
    this.phraseSampleSentence = phraseSampleSentence;
    this.phraseComment = phraseComment;
    this.translatedPhrase = translatedPhrase;
    this.translatedPhraseSampleSentence = translatedPhraseSampleSentence;
    this.translatedPhraseComment = translatedPhraseComment;
    this.languageLevel = languageLevel;
    this.category = category;
    this.trainLevel = trainLevel;
    this.markedAsHard = markedAsHard;
    this.lastTrainingDate = lastTrainingDate;
  }

  public initializeWithFlashcard(flashcard: FlashcardForLearn): void {
    this.flashcardId = flashcard.flashcardId;
    this.flashcardSubscriptionId = flashcard.flashcardSubscriptionId;
    this.phrase = flashcard.phrase;
    this.phrasePronunciation = flashcard.phrasePronunciation;
    this.phraseSampleSentence = flashcard.phraseSampleSentence;
    this.phraseComment = flashcard.phraseComment;
    this.translatedPhrase = flashcard.translatedPhrase;
    this.translatedPhraseSampleSentence = flashcard.translatedPhraseSampleSentence;
    this.translatedPhraseComment = flashcard.translatedPhraseComment;
    this.languageLevel = flashcard.languageLevel;
    this.category = flashcard.category;
    this.trainLevel = flashcard.trainLevel;
    this.markedAsHard = flashcard.markedAsHard;
    this.lastTrainingDate = flashcard.lastTrainingDate;
  }
}

File: FlashCards.Models/DTOs/ToClient/Learn/FlashcardForLearn.cs
using System;

namespace FlashCards.Models.DTOs.ToClient.Learn
{
    public class FlashcardForLearn
    {
        public int FlashcardId { get; set; }

        public int FlashcardSubscriptionId { get; set; }

        public string Phrase { get; set; }

        public string PhrasePronunciation { get; set; }

        public string PhraseSampleSentence { get; set; }

        public string PhraseComment { get; set; }

        public string TranslatedPhrase { get; set; }

        public string TranslatedPhraseSampleSentence { get; set; }

        public string TranslatedPhraseComment { get; set; }

        public string LanguageLevel { get; set; }

        public int TrainLevel { get; set; }

        public bool MarkedAsHard { get; set; }

        public DateTime LastTrainingDate { get; set; }


        public FlashcardForLearn() { }

        public FlashcardForLearn(FlashcardForLearn flashcardForLearn)
        {
            FlashcardId = flashcardForLearn.FlashcardId;
            FlashcardSubscriptionId = flashcardForLearn.FlashcardSubscriptionId;
            Phrase = flashcardForLearn.Phrase;
            PhrasePronunciation = flashcardForLearn.PhrasePronunciation;
            PhraseSampleSentence = flashcardForLearn.PhraseSampleSentence;
            PhraseComment = flashcardForLearn.PhraseComment;
            TranslatedPhrase = flashcardForLearn.TranslatedPhrase;
            TranslatedPhraseSampleSentence = flashcardForLearn.TranslatedPhraseSampleSentence;
            TranslatedPhraseComment = flashcardForLearn.TranslatedPhraseComment;
            LanguageLevel = flashcardForLearn.LanguageLevel;
            TrainLevel = flashcardForLearn.TrainLevel;
            MarkedAsHard = flashcardForLearn.MarkedAsHard;
            LastTrainingDate = flashcardForLearn.LastTrainingDate;
        }
    }
}
